onnxruntime/core/providers/nnapi/nnapi_builtin/builders/op_support_checker.cc

```diff
@@ -1183,35 +1155,35 @@ int UnaryOpSupportChecker::GetMinSupportedOpSet(const NodeUnit& node_unit) const
 }
 
 /* static */ bool UnaryOpSupportChecker::IsQuantizedOpSupported(
-    const InitializedTensorSet& initializers, const Node& node, const OpSupportCheckParams& params) {
-  const auto& op_type = node.OpType();
+    const InitializedTensorSet& initializers, const NodeUnit& node_unit, const OpSupportCheckParams& params) {
+  const auto& op_type = node_unit.OpType();
   ORT_ENFORCE(op_type == "QLinearSigmoid");
 
-  const auto& op_name = node.Name();
-  const auto input_defs(node.InputDefs());
-  // const auto output_defs(node.OutputDefs());
+  const auto& op_name = node_unit.Name();
 
-  if (input_defs.size() < 4)
+  // Check input scales and ZPs
+  if (!HasValidQuantizationScales(initializers, node_unit, {0}, params, true /* is_input */))
     return false;
-
-  bool has_output_zp = input_defs.size() == 5;
-
-  if (!HasValidQuantizationScales(initializers, node, {1, 3}, params))
+  if (!HasValidQuantizationZeroPoints(initializers, node_unit, {0}, true /* is_input */))
     return false;
```

Comment:
given the prevalence of this pattern:
```
if (!HasValidQuantizationScales(...)) return false;
if (!HasValidQuantizationZeroPoints(...)) return false;
```
would a helper function like `HasValidQuantizationScalesAndZeroPoints()` be useful?

Reply:
Given both `HasValidQuantizationScales` and `HasValidQuantizationZeroPoints` have similar structure but quite complex, this can be combined, can do it in next PR